core/SiteContentDetector.php

```diff
@@ -127,14 +141,14 @@ public function detectContent(array $detectContent = [SiteContentDetector::ALL_C
         }
 
         // No cache hit, no passed data, so make a request for the site content
```

Comment:
Just out of curiosity: Does it make sense to have the caching base on the idSite? Imho it would make more sense to do all the caching based on the main url of the site. That way the cache would be reused if more than one site is tracking the same url, and if the main url of a site would be changed, it would no longer use the cache.

Reply:
Especially if people do change the URL then it can be useful. Not sure how often that happens, and how often this happens before they start tracking where we mostly use it but this could be good indeed. 
We could make that like `$cacheKey = 'SiteContentDetector_' . md5($mainUrl);`?

Reply:
sounds fine